src/adapter/activateDaffodilDebug.ts

```diff
@@ -52,7 +52,18 @@ async function getFile(fileRequested, label, title) {
       })
       .then((fileUri) => {
         if (fileUri && fileUri[0]) {
-          return fileUri[0].fsPath
+          let path = fileUri[0].fsPath
+
+          if (
+            process.platform === 'win32' &&
+            path.charCodeAt(0) > 97 &&
+            path.charCodeAt(0) <= 122 &&
+            path.charAt(1) === ':'
```

Comment:
I'd move this check up 2 lines.

Reply:
Wouldn't that open us up to a NullPointerException? If fileUri[0] is null, it'll get caught by the if, and this code won't be hit.

Reply:
That if statement is nested in another if statement (line 54), so you won't get an NPE.  It might be a good idea to ensure that the length of the path is greater than 2 before peeking at different indexes (e.g, `charCodeAt(0)` and `charAt(1)`) without knowing the size of the path.  I instinctively look at those fail fast conditionals with what filters things the quickest and order those first.  If the second character is not a colon, then it saves you from the next 2 checks.

Reply:
I see what you're saying now - adding the length check is definitely a good idea, but I'm not seeing the reason for moving the `charAt(1) === ':'` above the drive letter check.

The path that makes it here comes from a file dialog, which should mean that it has been validated. For the short circuit to save us from the other checks, that validated path would have to have a lowercase ascii character as the first character and something other than a colon as the second character. Is that any more likely than a validated path that has a non-alphabetical character as the first character and a colon as the second?